stock_mts_mto_rule/models/stock_rule.py

```diff
@@ -104,4 +110,8 @@ def _run_split_procurement(self, procurements):
                 getattr(self.env["stock.rule"], "_run_%s" % rule.mto_rule_id.action)(
                     [(mto_procurement, rule.mto_rule_id)]
                 )
+            done_procurements_by_product.setdefault(procurement.product_id.id, 0)
+            done_procurements_by_product[
+                procurement.product_id.id
+            ] += procurement.product_qty
```

Comment:
Here with defaultdict it would look like.
```suggestion
            done_procurements_by_product[
                procurement.product_id.id
            ] += procurement.product_qty
```